web/src/stores/workflow-store.ts

```diff
@@ -87,6 +90,12 @@ type WorkflowStoreState = TReactFlowGraph & {
 	deleteNodeByIdx: (nodeIdx: number) => void;
 	deleteControlByIdx: (controlIdx: number) => void;
 	duplicateNodeByIdx: (nodeIdx: number) => void;
+	hasDeletedSecret: (nodeId: string, secretPlaceholer?: string) => boolean;
+	addDeletedSecrets: (missingSecrets: Map<string, Set<string>>) => void;
```

Comment:
This a setter, so the better naming would be `setDeletedSecrets`. Also, the parameter name mismatches the function name. `deletedSecrets` would probably be more appropriate.